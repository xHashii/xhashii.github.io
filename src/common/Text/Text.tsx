import React, { FC, HTMLAttributes } from "react";
import { Classes } from "jss";
import cx from "classnames";

export interface ITextProps extends HTMLAttributes<HTMLElement> {
  classes: Classes<"h1" | "h2" | "h3" | "caption" | "p">;
  children?: any;
  h1?: Boolean;
  h2?: Boolean;
  h3?: Boolean;
  caption?: Boolean;
  light?: Boolean;
}

const Text: FC<ITextProps> = ({
  classes = {},
  children = null,
  h1 = false,
  h2 = false,
  h3 = false,
  caption = false,
  className = '',
  ...props
}) => {
  if (h1) return <h1 className={cx(classes.h1, className)} {...props}>{ children }</h1>;

  if (h2) return <h2 className={cx(classes.h2, className)} {...props}>{ children }</h2>;

  if (h3) return <h3 className={cx(classes.h3, className)} {...props}>{ children }</h3>;

  if (caption) return <span className={cx(classes.caption, className)} {...props}>{ children }</span>;

  return <p className={cx(classes.p, className)} {...props}>{ children }</p>;
};

export default Text;
