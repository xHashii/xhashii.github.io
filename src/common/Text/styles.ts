import { Theme } from "../../utils/theme";

const defaultStyles = (theme: Theme) => ({
  color: theme.textLight,
  fontFamily: theme.fontFamily,
  fontWeight: 700,
  margin: 0,
  marginBottom: 4,
  textDecoration: 'none'
});

export default (theme: Theme) => {
  const defaults = defaultStyles(theme)

  return {
    h1: {
      ...defaults,
      fontSize: 48,
      marginBottom: 16,
      color: 'white'
    },
    h2: {
      ...defaults,
      fontSize: 36,
      marginBottom: 8
    },
    h3: {
      ...defaults,
      fontSize: 24,
      marginTop: 48
    },
    caption: {
      ...defaults,
      wordWrap: "break-word" as "break-word",
      fontSize: 12,
      letterSpacing: "0.08333333333333333em",
      marginBottom: 0,
      textTransform: "uppercase" as "uppercase",
      userSelect: 'none' as 'none',
      color: theme.colorActive,
    },
    p: {
      ...defaults,
      fontSize: 16,
      fontWeight: 400
    }
  };
};
