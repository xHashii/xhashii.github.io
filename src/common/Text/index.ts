import withSyles from "react-jss";

import Text from "./Text";
import styles from "./styles";

export default withSyles(styles)(Text);
