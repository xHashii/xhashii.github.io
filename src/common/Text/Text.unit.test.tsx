import React from "react";
import { shallow } from "enzyme";

import Text from "./Text";

const defaultProps = {
  classes: {}
};

describe("The Text instance", () => {
  describe("When rendered with default props", () => {
    it("should render a <p /> tag", () => {
      const wrapper = shallow(<Text bold {...defaultProps} />);
      const p = wrapper.find("p");

      expect(p.exists()).toBe(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when rendered with h1 set to true", () => {
    it("should render a <h1 /> tag", () => {
      const newProps = {
        ...defaultProps,
        h1: true
      };
      const wrapper = shallow(<Text {...newProps} />);
      const h1 = wrapper.find("h1");

      expect(h1.exists()).toBe(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when rendered with h2 set to true", () => {
    it("should render a <h2 /> tag", () => {
      const newProps = {
        ...defaultProps,
        h2: true
      };
      const wrapper = shallow(<Text {...newProps} />);
      const h2 = wrapper.find("h2");

      expect(h2.exists()).toBe(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when rendered with h3 set to true", () => {
    it("should render a <h3 /> tag", () => {
      const newProps = {
        ...defaultProps,
        h3: true
      };
      const wrapper = shallow(<Text {...newProps} />);
      const h3 = wrapper.find("h3");

      expect(h3.exists()).toBe(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when rendered with caption set to true", () => {
    it("should render a <caption /> tag", () => {
      const newProps = {
        ...defaultProps,
        caption: true
      };
      const wrapper = shallow(<Text {...newProps} />);
      const caption = wrapper.find("span");

      expect(caption.exists()).toBe(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when style overrides are passed", () => {
    it("should add a style attribute to the element with the overrides", () => {
      const newProps = {
        ...defaultProps,
        style: {
          fontSize: 0
        }
      };
      const wrapper = shallow(<Text {...newProps} />);
      const p = wrapper.find("p");

      expect(p.props().style).toEqual(newProps.style);
    });
  });
});
