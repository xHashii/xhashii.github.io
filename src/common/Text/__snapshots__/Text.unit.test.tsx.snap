// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Text instance When rendered with default props should render a <p /> tag 1`] = `ShallowWrapper {}`;

exports[`The Text instance when rendered with caption set to true should render a <caption /> tag 1`] = `ShallowWrapper {}`;

exports[`The Text instance when rendered with h1 set to true should render a <h1 /> tag 1`] = `ShallowWrapper {}`;

exports[`The Text instance when rendered with h2 set to true should render a <h2 /> tag 1`] = `ShallowWrapper {}`;

exports[`The Text instance when rendered with h3 set to true should render a <h3 /> tag 1`] = `ShallowWrapper {}`;
